packages/uiweb/src/lib/components/widget/subscriptionManager/SubscribeComponent.tsx

```diff
@@ -72,70 +61,100 @@ export const SubscribeComponent: React.FC<ISubscribeComponentProps> = (
         toastType: 'ERROR',
-        getToastIcon: (size) => <MdError size={size} color="red" />,
+        getToastIcon: (size) => (
+          <MdError
+            size={size}
+            color="red"
+          />
+        ),
       });
     }
   };
   return (
-    <Section flexDirection="column" gap="10px" margin="14px 10px 0px 10px">
+    <Section
+      flexDirection="column"
+      gap="10px"
+      margin="14px 10px 0px 10px"
+    >
+      <Section
+        flexDirection="column"
+        alignItems="start"
+        gap="5px"
+        margin="0 0 10px 0"
+      >
+        <Span
+          fontSize="21px"
+          fontWeight="700"
+          color={theme?.textColor?.modalHeaderText}
+        >
+          Subscribe to get Notified
+        </Span>
+        <Span
+          fontSize="12px"
+          textAlign="left"
+          fontWeight="400"
+          width="90%"
+          color={theme?.textColor?.modalSubTitleText}
+        >
+          {SubHeadingText.withoutSettings}
+        </Span>
+      </Section>
+      <ChannelDetailsComponent channelInfo={channelInfo} />
+      {channelInfo && channelInfo?.channel_settings && (
         <Section
-          flexDirection="column"
-          alignItems="start"
-          gap="5px"
-          margin="0 0 10px 0"
+          margin=" 0"
+          width="100%"
         >
-          <Span fontSize="21px" fontWeight="700" color={theme?.textColor?.modalHeaderText}>
-            Subscribe to get Notified
-          </Span>
-          <Span
-            fontSize="12px"
-            textAlign="left"
-            fontWeight="400"
-            width="90%"
-            color={theme?.textColor?.modalSubTitleText}
-          >
-            {SubHeadingText.withoutSettings}
-          </Span>
+          <SettingsComponent
+            settings={modifiedSettings!}
+            setSettings={setModifiedSettings}
+          />
         </Section>
-        <ChannelDetailsComponent channelInfo={channelInfo}/>
-     {channelInfo && channelInfo?.channel_settings && <Section margin=' 0' width='100%' >
-      <SettingsComponent settings={modifiedSettings!} setSettings={setModifiedSettings}/>
-      </Section>}
-      
-   {(user && !!user?.readmode)?
-   <>
-     {user?.readmode() && (
-        <ConnectButtonComp
-          autoconnect={autoconnect}
-          setAccount={setAccount}
-          setSigner={setSigner}
-          signer={signer}
-        />)}
-         
-       {!user?.readmode() &&  
-      <Button onClick={handleSubscribe}>
-        {subscribeLoading ? <Spinner color="#fff" size="24" /> : 'Subscribe'}
-      </Button>}
-      </>
-     :null
-    }
- 
+      )}
+
+      {user && !!user?.readmode ? (
+        <>
+          {user?.readmode() && (
+            <ConnectButton
+              autoconnect={autoconnect}
+              setAccount={setAccount}
+              setSigner={setSigner}
+              signer={signer}
+            />
+          )}
+
+          {!user?.readmode() && (
```

Comment:
We are using both user.readmode and user.readmode() as variable and as a function, which one is correct

Reply:
We are checking if the user had readmode function then we will call it int he below line @rohitmalhotra1420